tracing-subscriber/src/fmt/format/pretty.rs

```diff
@@ -342,27 +347,42 @@ impl<'a> field::Visit for PrettyVisitor<'a> {
         if self.result.is_err() {
             return;
         }
-        let bold = self.bold();
-        match field.name() {
-            "message" => self.write_padded(&format_args!("{}{:?}", self.style.prefix(), value,)),
-            // Skip fields that are actually log metadata that have already been handled
-            #[cfg(feature = "tracing-log")]
-            name if name.starts_with("log.") => self.result = Ok(()),
-            name if name.starts_with("r#") => self.write_padded(&format_args!(
-                "{}{}{}: {:?}",
-                bold.prefix(),
-                &name[2..],
-                bold.infix(self.style),
-                value
-            )),
-            name => self.write_padded(&format_args!(
-                "{}{}{}: {:?}",
-                bold.prefix(),
-                name,
-                bold.infix(self.style),
-                value
-            )),
-        };
+        if self.writer.has_ansi_escapes() {
+            let bold = self.bold();
+            match field.name() {
+                "message" => {
+                    self.write_padded(&format_args!("{}{:?}", self.style.prefix(), value,))
+                }
+                // Skip fields that are actually log metadata that have already been handled
+                #[cfg(feature = "tracing-log")]
+                name if name.starts_with("log.") => self.result = Ok(()),
+                name if name.starts_with("r#") => self.write_padded(&format_args!(
+                    "{}{}{}: {:?}",
+                    bold.prefix(),
+                    &name[2..],
+                    bold.infix(self.style),
+                    value
+                )),
+                name => self.write_padded(&format_args!(
+                    "{}{}{}: {:?}",
+                    bold.prefix(),
+                    name,
+                    bold.infix(self.style),
+                    value
+                )),
+            };
+        } else {
+            match field.name() {
+                "message" => self.write_padded(&format_args!("{:?}", value)),
+                // Skip fields that are actually log metadata that have already been handled
+                #[cfg(feature = "tracing-log")]
+                name if name.starts_with("log.") => self.result = Ok(()),
+                name if name.starts_with("r#") => {
+                    self.write_padded(&format_args!("{}: {:?}", &name[2..], value))
+                }
+                name => self.write_padded(&format_args!("{}: {:?}", name, value)),
+            };
+        }
```

Comment:
note that you shouldn't actually need to duplicate this code: `PrettyVisitor::bold` will return a no-op style (which doesn't emit any ANSI color codes) in the case where the writer does not have ANSI escapes enabled:
https://github.com/tokio-rs/tracing/blob/c90d22b02380c57a455ff3cf721f19d697bcd090/tracing-subscriber/src/fmt/format/pretty.rs#L304-L311
So, we could actually just write this:
```suggestion
        let bold = self.bold();
        match field.name() {
            "message" => {
                self.write_padded(&format_args!("{}{:?}", self.style.prefix(), value,))
            }
            // Skip fields that are actually log metadata that have already been handled
            #[cfg(feature = "tracing-log")]
            name if name.starts_with("log.") => self.result = Ok(()),
            name if name.starts_with("r#") => self.write_padded(&format_args!(
                "{}{}{}: {:?}",
                bold.prefix(),
                &name[2..],
                bold.infix(self.style),
                value
            )),
            name => self.write_padded(&format_args!(
                "{}{}{}: {:?}",
                bold.prefix(),
                name,
                bold.infix(self.style),
                value
            )),
        };
```